src/Keboola/DbExtractor/Extractor/Extractor.php

```diff
@@ -220,6 +228,12 @@ private function handleDbError(Throwable $e, ?array $table = null, ?int $counter
         return new UserException($message, 0, $e);
     }
 
+    protected function isAlive(): void
+    {
+        // needs to be implemented in extractors.
```

Comment:
shouldn't it be abstract then? Also better API would probably be to have isAlive() that internally calls something like `abstract protected function checkIfAlive(): bool` and throw the exception itself? As there is no way to make sure they will throw the exception it would be safer to make them implement true/false (typehintable) method and handle the exception yourself here.

Reply:
yes, but I don't want to introduce any BC break:(

Reply:
but you are right about how the interface should be, there's no way to mandate a throw.

Reply:
So what about wrapping the exception check in isAlive in `if (method_exists($this, 'checkIfAlive')) ...` and returning true otherwise. It would be progressive enhancement.

Reply:
i don't know, can it be made without a BC break? if the specific extractor doesn't implment the isAlive method, it won't work either, so it's a BC break in either case

Reply:
ah, I see what @tomasfejfar is saying.
But I don't really like the `bool` return for checkIfAlive().  imho we want to propagate the native error.

Reply:
`: ?\Throwable`? :)

Reply:
how about we add it to https://github.com/keboola/db-extractor-common/issues/70?

Reply:
the question is - what is going to happen if we update e.g. mysql-ex to this version and it doesn't have the isAlive method ? i think that something will break -> the method should be required -> it should part of the interface -> it's a BC break

personally i don't care whether it's version 10.x.x or 20.x.x

Reply:
If mysql-ex doesn't implement it it won't break.  that method will simply do nothing and it will behave as before.

Reply:
but sure,  I'll succumb to pressure if that's what you guys agree is best:)

Reply:
lol fark,  why didn't I just use the already present `testConnection()`  :p

Reply:
there, now nothing needs to be implemented 👍

Reply:
because it doesn't have the necessary interface? :) ok, let's leave it as it it - use testConnection and fix the interface in #70 if needed

Reply:
If you're ok with that @tomasfejfar I'll go ahead and merge and release mysql